pkg/prometheus/agent/operator.go

```diff
@@ -1373,3 +1418,66 @@ func (c *Operator) getMissingPermissions(ctx context.Context) (map[string][]stri
 
 	return missingPermissions, nil
 }
+
+// checkCRDAuthorization checks if the operator has access to the specified CRD
+// It checks individual namespaces if an allowlist was provided, otherwise it checks cluster-wide.
+func (c *Operator) checkCRDAuthorization(ctx context.Context, verb string, group string, resource string) (bool, error) {
```

Comment:
I am moving them to k8sutil package in https://github.com/prometheus-operator/prometheus-operator/pull/5476 :)

Reply:
`getMissingPermissions` is definitely similar, I had taken what you had done in the Agent PR for this one 😅 I think this can be a good follow up once #5476 is merged.